src/dmd/dsymbolsem.d

```diff
@@ -3403,7 +3403,13 @@ private extern(C++) final class DsymbolSemanticVisitor : Visitor
          */
         if (f.isscope && !funcdecl.isNested() && !ad)
         {
-            funcdecl.error("functions cannot be `scope`");
+            /* Non-static nested functions have a hidden 'this' pointer to which
+             * the 'scope' applies
+             */
+            if (sc.scopesym && sc.scopesym.isAggregateDeclaration())
+                funcdecl.error("`static` member has no `this` to which `scope` can apply");
+            else
+                error(funcdecl.loc, "Top-level function `%s` has no `this` to which `scope` can apply", funcdecl.toChars());
```

Comment:
This was modified to emit a similar error message `return` a few lines below.  Please expand the source code to see.